broker-core/src/main/java/io/zeebe/broker/incident/processor/TypedWorkflowInstanceRecord.java

```diff
@@ -26,16 +26,18 @@
 
 public class TypedWorkflowInstanceRecord implements TypedRecord<WorkflowInstanceRecord> {
 
-  private final IndexedRecord persistedWorkflowInstanceRecord;
   private final RecordMetadata recordMetadata;
 
-  public TypedWorkflowInstanceRecord(
-      long incidentKey, IndexedRecord persistedWorkflowInstanceRecord) {
-    this.persistedWorkflowInstanceRecord = persistedWorkflowInstanceRecord;
+  private IndexedRecord persistedWorkflowInstanceRecord;
 
+  public TypedWorkflowInstanceRecord() {
     recordMetadata = new RecordMetadata();
     recordMetadata.recordType(RecordType.EVENT);
     recordMetadata.valueType(ValueType.WORKFLOW_INSTANCE);
+  }
+
+  public void wrap(long incidentKey, IndexedRecord persistedWorkflowInstanceRecord) {
+    this.persistedWorkflowInstanceRecord = persistedWorkflowInstanceRecord;
     recordMetadata.intent(persistedWorkflowInstanceRecord.getState());
     recordMetadata.incidentKey(incidentKey);
```

Comment:
I think that we need the `incidentKey` in the metadata no longer because you removed the `RESOLVE_FAILED` intent. We should get rid of it ;)

Reply:
But with this we see which following event belongs to which resolved incident. Maybe interessting ? :smile:

Reply:
ok done